src/Extensions/AzureBlobPayloads/PayloadStore/BlobPayloadStore.cs

```diff
@@ -0,0 +1,165 @@
+/// Stores payloads as blobs and returns opaque tokens in the form "blob:v1:&lt;container&gt;:&lt;blobName&gt;".
+/// </summary>
+sealed class BlobPayloadStore : IPayloadStore
+{
+    const string TokenPrefix = "blob:v1:";
+    const string ContentEncodingGzip = "gzip";
+    const int DefaultCopyBufferSize = 81920;
+    const int MaxRetryAttempts = 8;
+    const int BaseDelayMs = 250;
+    const int MaxDelayMs = 10_000;
+    const int NetworkTimeoutMinutes = 2;
+    readonly BlobContainerClient containerClient;
+    readonly LargePayloadStorageOptions options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="BlobPayloadStore"/> class.
+    /// </summary>
+    /// <param name="options">The options for the blob payload store.</param>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="options"/> is null.</exception>
+    /// <exception cref="ArgumentException">Thrown when neither connection string nor account URI/credential are provided.</exception>
+    public BlobPayloadStore(LargePayloadStorageOptions options)
+    {
+        this.options = options ?? throw new ArgumentNullException(nameof(options));
+        Check.NotNullOrEmpty(options.ContainerName, nameof(options.ContainerName));
+
+        // Validate that either connection string or account URI/credential are provided
+        bool hasConnectionString = !string.IsNullOrEmpty(options.ConnectionString);
+        bool hasIdentityAuth = options.AccountUri != null && options.Credential != null;
+
+        if (!hasConnectionString && !hasIdentityAuth)
+        {
+            throw new ArgumentException(
+                "Either ConnectionString or AccountUri and Credential must be provided.",
+                nameof(options));
+        }
+
+        BlobClientOptions clientOptions = new()
+        {
+            Retry =
+            {
+                Mode = RetryMode.Exponential,
+                MaxRetries = MaxRetryAttempts,
+                Delay = TimeSpan.FromMilliseconds(BaseDelayMs),
+                MaxDelay = TimeSpan.FromSeconds(MaxDelayMs),
+                NetworkTimeout = TimeSpan.FromMinutes(NetworkTimeoutMinutes),
+            },
+        };
+
+        BlobServiceClient serviceClient = hasIdentityAuth
+            ? new BlobServiceClient(options.AccountUri, options.Credential, clientOptions)
+            : new BlobServiceClient(options.ConnectionString, clientOptions);
+
+        this.containerClient = serviceClient.GetBlobContainerClient(options.ContainerName);
+    }
+
+    /// <inheritdoc/>
+    public override async Task<string> UploadAsync(ReadOnlyMemory<byte> payloadBytes, CancellationToken cancellationToken)
+    {
+        // One blob per payload using GUID-based name for uniqueness (stable across retries)
+        string blobName = $"{Guid.NewGuid():N}";
+        BlobClient blob = this.containerClient.GetBlobClient(blobName);
+
+        byte[] payloadBuffer = payloadBytes.ToArray();
```

Comment:
This creates a temporary in-memory copy of a potentially large chunk of data. Since this data remains read-only, perhaps there is a way to get rid of this copy. `Stream.WriteAsync` accepts `ReadOnlyMemory<byte>` directly, so you may be able to *also* skip wrapping this data into a `MemoryStream` and just do `blobStream.WriteAsync(payloadBytes, ...)` or `compressedBlobStream(payloadBytes, ...)`.